File: 2025-04-April-GFG-POTD/April 24 - Unique Number III/Solution.py
class Solution:
    def getSingle(self, arr):
        freq = {}
        for num in arr:
            freq[num] = freq.get(num, 0) + 1
        
        for key, value in freq.items():
            if value == 1:
                return key
        return -1  # If not found

# Driver Code
t = int(input())
for _ in range(t):
    arr = list(map(int, input().split()))
    sol = Solution()
    print(sol.getSingle(arr))
    print("~")

File: 2025-04-April-GFG-POTD/April 24 - Unique Number III/Solution.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int getSingle(vector<int>& arr) {
        unordered_map<int, int> freq;

        for (int num : arr) {
            freq[num]++;
        }

        for (auto& entry : freq) {
            if (entry.second == 1) {
                return entry.first;
            }
        }

        return -1; // If not found
    }
};

int main() {
    int t;
    cin >> t;
    cin.ignore();

    while (t--) {
        string line;
        getline(cin, line);
        vector<int> arr;
        int num;
        stringstream ss(line);

        while (ss >> num) {
            arr.push_back(num);
        }

        Solution sol;
        int ans = sol.getSingle(arr);
        cout << ans << endl;
        cout << "~" << endl;
    }

    return 0;
}
